src/lib/utils.ts

```diff
@@ -178,3 +217,52 @@ export function Operations<M extends Model>(
 
   return { applyPropertyExpression, onAggBuild };
 }
+
+/**
+ * Determines if a property is a [FieldExpression](https://vincit.github.io/objection.js/api/types/#type-fieldexpression)
+ * @param property The property to check
+ */
+export function isFieldExpression(property: string): boolean {
+  return property.indexOf('$') > -1;
+}
+
+/**
+ * Builds a reference for a FieldExpression with support for fully-qualified properties
+ * @param property a FieldExpression string
+ */
+export function getFieldExpressionRef(property: string): ReferenceBuilder {
+  const isFullyQualified = property.indexOf(':') > -1;
+  if (isFullyQualified) {
+    let { propertyName, relationName } = sliceRelation(property, ':');
+    relationName = relationName.replace('.', ':');
+    propertyName = propertyName.replace('$', ':');
+    return (ref(propertyName) as any).from(relationName);
+  }
+
+  const propertyName = property.replace('$', ':');
+  return ref(propertyName);
+}
+
+/**
+ * Casts a ReferenceBuilder instance to a type based on the type of the operand
+ * @param reference A reference built from a filterExpression
+ * @param operand the operand from which to infer the type
+ */
+export function castTo(
+  reference: ReferenceBuilder,
+  operand: Expression
+): ReferenceBuilder {
+  const type = typeof operand;
+  if (type === 'string') {
+    return reference.castText();
+  }
+  if (type === 'boolean') {
+    return reference.castBool();
+  }
+  if (type === 'number') {
+    return reference.castDecimal();
+  }
+
+  // Don't cast by default
+  return reference;
```

Comment:
Not really sure how this branch of this function can be tested, since JSON primitives will mean that execution never gets to here. Although this should be fine as a fallback anyway.